From 8be9fdf9552faa75753c1ef63451a39876bbffdb Mon Sep 17 00:00:00 2001
From: Gu Zitao <guzitao@wxiat.com>
Date: Mon, 7 Nov 2022 14:17:32 +0800
Subject: [PATCH] sw64: mark sched_clock() as notrace
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 8be9fdf9552faa75753c1ef63451a39876bbffdb
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I645RC

--------------------------------

sched_clock() of SW64 has no notrace mark like the generic declaration
and it will cause kernel crash in ftrace selftests. Add notrace to fix
it.

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Reviewed-by: He Sheng <hesheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/time.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/kernel/time.c b/arch/sw_64/kernel/time.c
index 32690e78849b..8be7a11df51d 100644
--- a/arch/sw_64/kernel/time.c
+++ b/arch/sw_64/kernel/time.c
@@ -143,7 +143,7 @@ void __init sw64_sched_clock_init(void)
 	sched_clock_register(sched_clock_read, BITS_PER_LONG, get_cpu_freq() >> sc_shift);
 }
 #else
-unsigned long long sched_clock(void)
+unsigned long long notrace sched_clock(void)
 {
 	if (static_branch_likely(&use_tc_as_sched_clock))
 		return ((rdtc() - sc_start + __this_cpu_read(tc_offset)) >> sc_shift) * sc_multi;
-- 
2.33.0

